packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -43,23 +46,37 @@ const DataGridPremiumRaw = React.forwardRef(function DataGridPremium<R extends G
 
   useLicenseVerifier('x-data-grid-premium', releaseInfo);
 
+  const { pivotParams } = props;
+
+  const hasPivotModel = pivotParams?.pivotModel != null;
+
   if (process.env.NODE_ENV !== 'production') {
     validateProps(props, dataGridPremiumPropValidators);
   }
   return (
     <GridContextProvider privateApiRef={privateApiRef} props={props}>
       <GridRoot
-        className={props.className}
+        className={clsx(props.className, hasPivotModel && gridClasses.sidePanel)}
```

Comment:
This class is only used for the flex style. We could just apply it by default to the root class, simpler & it also avoid exposing a new class in the public API.